OneCommunityGlobal/HighestGoodNetworkApp
# Reina delete own time

# Description
This is a continuation of PR#2087

"This PR enables the user with the Delete Time Entry (Own) permission to be able to delete it's own any time entry via their dashboard."

## Related PRS (if any):
This frontend PR is related to the dev backend PR.
…

## Main changes explained:
- Fixed a few bugs in PR2087. 
- Added some conditional logic to a few files to enable own permissions.
…

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. Clear site data/cache
4. log as admin user
5. go to Other Links -> Permissions Management -> Manage User Permissions -> Choose a USER without Delete Time Entry (Own) permission under Timelog Management -> click on Add -> Submit
6. Make sure USER doesn't have DELETE_TIME_ENTRY_OTHERS Permission
7. Verify that the user who was given permission is now able to see the delete button in any of time entry log on the dashboard and is able to delete it.

## Screenshots or videos of changes:


https://github.com/user-attachments/assets/d6e66ebe-d709-46e5-8171-a77379f85fa6
